lib/Cake/Controller/Component/SecurityComponent.php

```diff
@@ -344,7 +344,7 @@ public function requireAuth() {
  * @link https://book.cakephp.org/2.0/en/core-libraries/components/security-component.html#handling-blackhole-callbacks
  * @throws BadRequestException
  */
-	public function blackHole(Controller $controller, $error = '', SecurityException $exception = null) {
+	public function blackHole(?Controller $controller, $error = '', ?SecurityException $exception = null) {
```

Comment:
Shouldn't we keep `Controller`?

Reply:
Yes, $controller is not optional, my bad. Thank you!
(and I also corrected a typo that I just spotted)

Reply:
Thanks for fix :raised_hands: